config/v1/types_image.go

```diff
@@ -112,4 +112,9 @@ type RegistrySources struct {
 	// Only one of BlockedRegistries or AllowedRegistries may be set.
 	// +optional
 	AllowedRegistries []string `json:"allowedRegistries,omitempty"`
+	// containerRuntimeSearchRegistries are registries that will be searched when pulling images that do not have fully qualified
+	// domains in their pull specs. Registries will be searched in the order provided in the list.
+	// Note: this search list only works with the container runtime, i.e CRI-O. Will NOT work with builds or imagestream imports.
+	// +optional
+	ContainerRuntimeSearchRegistries []string `json:"containerRuntimeSearchRegistries,omitempty"`
```

Comment:
which format do these have? Please add a specification (OpenAPI pattern, format, minLength, unique; https://book.kubebuilder.io/reference/markers/crd-validation.html).

Are these unique? Can there be duplicates? Is this a set? Compare https://book.kubebuilder.io/reference/markers/crd-processing.html?highlight=listtype#crd-processing `ListType`.

Reply:
There are no specific length or format requirements here. There can be duplicates in the list. This list already exists on the node in `/etc/containers/registries.conf`, exposing this helps the user configure the list on the node. This new option is similar to the existing options under `RegistrySources` currently.
Do we still need to add specifications in this case?

Reply:
Yes, as part of a spec these are supposed to be fully specified. Adding restriction later is hard. So we have to be precise from the beginning.

I am pretty sure these cannot be empty string. And they are URLs? And this looks like a set.

Reply:
Got it, added the restrictions that I believe should apply to this field. PTAL

Reply:
@sttts is this good now, PTAL